### PR TITLE
add docker registry for kind Kubernetes cluster

### PR DESCRIPTION
**Pull request by xiaojingchen**
### What problem does this PR solve? 
currently, `hack/kind-cluster-build.sh` has not provided inner docker registry for user
### What is changed and how does it work?
add docker registry for kind Kubernetes cluster
### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has documents change

Side effects

Related changes

### Does this PR introduce a user-facing change?:
 
 ```release-note
NONE
 ```

### CODE REVIEW

File: hack/kind-cluster-build.sh [hunk 1]
```diff
@@ -119,6 +125,137 @@ echo "start to create k8s cluster"
...
+                operator: NotIn
+                values:
+                  - ${registryNode}
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        operator: "Equal"
+        effect: "NoSchedule"
+      containers:
+        - name: nginx
+          image: nginx:1.15-alpine
```
Comment (cofyc) [inline hack/kind-cluster-build.sh]: register:2 has builtin http server, what's the purpose of nginx proxy?

Comment (xiaojingchen) [reply hack/kind-cluster-build.sh]: Because the kubelets use localhost:5000 to get images, so need to deploy `nginx-proxy` on every node.

Comment (tennix) [reply hack/kind-cluster-build.sh]: Only the localhost is trusted by Docker or containerd. Otherwise, we have to configure TLS for the registry.

Comment (cofyc) [reply hack/kind-cluster-build.sh]: I see, this makes sense

Comment (cofyc) [reply hack/kind-cluster-build.sh]: BTW, is a plain TCP proxy tunneling enough here? if it is, use socat (docker [image](https://hub.docker.com/r/alpine/socat/)) is much simpler, e.g. `socat tcp-listen:5000,fork,reuseaddr tcp-connect:<registry2-ip>:5000`.

**Pull request merged**